csrc/fusion.cpp

```diff
@@ -766,6 +769,10 @@ bool Fusion::isAliasCompatible(Val* left, Val* right) {
 }
 
 void Fusion::aliasOutputToInput(Val* output, Val* input, const AliasType type) {
+  NVF_CHECK(
+      type != AliasType::NoAlias,
+      "NoAlias is returned automatically for a missing key. Don't add it explicitly.");
+
```

Comment:
OOC, under what situations can we fail this check?

Reply:
If some code in the future mistakenly call `aliasOutputToInput` with `type=AliasType::NoAlias`.